python/ray/tune/logger.py

```diff
@@ -119,6 +122,41 @@ def to_tf_values(result, path):
     return values
 
 
+class UtilMonitor(Thread):
+    def __init__(self, delay):
+        super(UtilMonitor, self).__init__()
+        self.stopped = False
+        self.delay = delay  # Time between calls to GPUtil
+        self.values = defaultdict(list)
+        self.lock = threading.Lock()
+        self.start()
+
+    def read_utilization(self):
+        self.lock.acquire()
```

Comment:
Use `with self.lock:` block instead of `acquire` and `release`.

Reply:
nice touch!